Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPILocalizationCocoa.mm

```diff
@@ -88,6 +88,16 @@
     callback->call(acceptLanguages.array);
 }
 
+void WebExtensionAPILocalization::getPreferredSystemLanguages(Ref<WebExtensionCallbackHandler>&& callback)
+{
+    callback->call(NSLocale.preferredLanguages);
+}
+
+void WebExtensionAPILocalization::getSystemUILanguage(Ref<WebExtensionCallbackHandler>&& callback)
+{
+    callback->call(NSLocale.systemLocale.localeIdentifier);
```

Comment:
This will give the web extension format, the system string can have additional stuff added.
```suggestion
    callback->call(toWebAPI(NSLocale.systemLocale));
```

Reply:
@xeenon Thanks! Implemented this change.

Reply:
@xeenon Seems this change makes it nolonger cast to a string if undefined. My proposal would be to, in the case systemLocale is not present or can not be provided, to resolve to "und" or null. I can update the PR to reflect this.

Reply:
Currently `toWebAPI()` returns JS `undefined` when it is `nil`. Can can update that to return `und` is that is more inline with what Chrome/Firefox does / will do. That method is in `WebExtensionUtilities.mm`.

```objc++
NSString *toWebAPI(NSLocale *locale)
{
    if (!locale.languageCode)
        return nil;

    if (locale.countryCode.length)
        return [NSString stringWithFormat:@"%@-%@", locale.languageCode, locale.countryCode];
    return locale.languageCode;
}
```

Just change `return nil` there to `return @"und"`.

Reply:
For `null` you would return `NSNull.null` and change the return type to `id`.

Reply:
Why are these cases returning no language in the tests?